packages/@aws-cdk/aws-lambda/lib/alias.ts

```diff
@@ -193,6 +201,39 @@ export class Alias extends QualifiedFunctionBase implements IAlias {
     });
   }
 
+  /**
+   * Enable autoscaling for the given lambda version. Requires that lambda version has provisioned concurrency.
+   * @param props The properties for autoscaling
+   */
+  public autoScaleProvisionedConcurrency(props: EnableScalingProps): IScalableVersionAttribute {
+    if (!this.provisionedConcurrency) {
+      throw new Error('Autoscaling is available for aliases with provisioned concurrency only');
+    }
+    if (this.scalableAlias) {
+      throw new Error('Autoscaling already enabled for this alias');
+    }
+    // Use a Service Linked Role
+    // https://docs.aws.amazon.com/autoscaling/application/userguide/application-auto-scaling-service-linked-roles.html
+    const role = iam.Role.fromRoleArn(this, 'ScalingRole', Stack.of(this).formatArn({
```

Comment:
Done -- but wouldn't calling the method a second time on the same alias result in the error: 'Autoscaling already enabled on this alias'. I figured it would be bad if the user could create multiple autoscaling plans on the same alias, which is why that error is there.

Reply:
Even if we don't want to allow that, meaning we want to prevent the user from creating several autoscaling configuration, a better way would be to make this function idempotent, the question is, would a customer want to call this method more than once, or will it only be accidental? If the answer is - "a customer might want to call this function more than once, and we don't support it", then we should throw an error, if not then we should ignore it. Either way we should be intentional about error handling, the error thrown from creating a resource with the same ID is not informative for the user

Reply:
I feel like a customer might accidentally call this method more than once but will never want two `scalableAlias` resources competing against each other. The customer might call the method twice when in reality they want two scaling schedules, for example. Creating two scaling schedules can be achieved by calling `scaleOnSchedule()` twice from `scalableAlias`. 

Given what you lay out, the answer would be to ignore the second call to the method. I've updated it accordingly.

But to be clear, the error handling would be `throw new Error('Autoscaling already enabled for this alias');` which is a bit more descriptive than the error thrown from creating a resource with the same ID. So there was some intentional error handling here to begin with.

Reply:
So on second thought, @NetaNir, the customer might want to call `autoScaleProvisionedConcurrency()` a second time with different `minCapacity` and `maxCapacity` numbers. I think it would be erroneous on our part to ignore this second call and return the original `scalableAlias` with the original min/max capacity. In light of this, I've kept the error of `Autoscaling already enabled for this alias` to alert the customer that PC autoscaling exists and that they can call `onSchedule` or `onUtilization` to configure their autoscaling.

Reply:
>So on second thought, @NetaNir, the customer might want to call `autoScaleProvisionedConcurrency()` a second time with different `minCapacity` and `maxCapacity` numbers. 

Why? What is the use case?

Reply:
Maybe not a use case, but if this happens:

```ts
alias.autoScaleProvisionedConcurrency({ minCapacity: 10, maxCapacity: 30});
...
alias.autoScaleProvisionedConcurrency({ minCapacity: 40, maxCapacity: 60});
```

Right now, it would throw the error `Autoscaling already enabled for this alias`. I feel like this is a mistake by the customer and this should be called out to them.

The other option would be to ignore the second call and keep the minCapacity/maxCapacity from the first call. I feel like this would be misleading for the user, who might think that `minCapacity == 40` now when in reality it still is ` == 10`.

Perhaps a use case could be that the user wants to update the minCapacity/maxCapacity of the scalingTarget, which IS possible, but is not supported in the CDK.

Anyway, it just doesn't feel right to ignore the parameters of the second autoScaling call and just return the original scaling target.